lib/puppet/network/http/api.rb

```diff
@@ -6,4 +6,27 @@ def self.not_found
         raise Puppet::Network::HTTP::Error::HTTPNotFoundError.new("No route for #{req.method} #{req.path}", Puppet::Network::HTTP::Issues::HANDLER_NOT_FOUND)
       end)
   end
+
+  def self.master_routes
+    master_prefix = Puppet[:master_url_prefix]
+    if !master_prefix.end_with?("/")
```

Comment:
There are cases (such as when we insert it into the default ACLs, and in all the tests) where we don't want it to have a trailing slash. It wouldn't be that much work to set it in the setting hook and change all the other instances, but it was less work to do this.